pkg/cmd/config/set-garden.go

```diff
@@ -0,0 +1,117 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/component-base/cli/flag"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+
+	"github.com/spf13/cobra"
+)
+
+// NewCmdConfigSetGarden returns a new (config) set-garden command.
+func NewCmdConfigSetGarden(f util.Factory, o *SetGardenOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set-garden",
+		Short: "modify or add Garden to gardenctl configuration",
+		Long:  "Modify or add Garden to gardenctl configuration. E.g. \"gardenctl config set-garden my-garden --kubeconfig ~/.kube/kubeconfig.yaml\" to configure or add a garden with identity 'my-garden'",
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			suggestions, err := validGardenArgsFunction(f, args)
+			if err != nil {
+				fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+
+			return util.FilterStringsByPrefix(toComplete, suggestions), cobra.ShellCompDirectiveNoFileComp
+		},
+		RunE: base.WrapRunE(o, f),
+	}
+
+	cmd.Flags().Var(&o.KubeconfigFile, "kubeconfig", "path to kubeconfig file for this Garden cluster")
+	cmd.Flags().Var(&o.ContextName, "context", "override the current-context of the garden cluster kubeconfig")
```

Comment:
Why not StringVar?

Reply:
We need the type StringFlag to determine if a flag has been provided. Otherwise it would not be possible to overwrite only parts of the struct. If you have any other idea to achieve this, please let me know. However as kubectl also uses it for this purpose I think it is valid to do so.